data/lavezzo2020suppression/lavezzo2020suppression-extraction.md

```diff
@@ -0,0 +1,250 @@
+df_test = df_demo.loc[:,datetime.datetime(2020, 2, 21, 0, 0):datetime.datetime(2020, 3, 10, 0, 0)].join(df_demo['id'])
+df_test = pd.melt(df_test,id_vars=['id'],var_name = 'test_date',value_name = 'results')
+df_test = df_test[~df_test['results'].isnull()]
+df_test['results'].value_counts()
+
+```
+
+```python
+#Filter to id with available PCR results.
+df_test = df_test[df_test['id'].isin(df['id'])]
+#Convert to date
+df_test['test_date'] = pd.to_datetime(df_test['test_date'])
+# df_test['test'] = df_test['test_date'].dt.date
+```
+
+```python
+df_test['test_date'].value_counts()
+```
+
+```python
+df_test_1= df_test[df_test['test_date'] < datetime.datetime(2020, 3, 6)]
+df_test_2 = df_test[df_test['test_date'] >= datetime.datetime(2020, 3, 6)]
+```
+
+```python
+#Generate the first positive /neg day for both first and second round
+    
+for type in ['Pos','Neg']:
+    temp_m = pd.DataFrame()
+    for round in range(1,3):
+        if round == 1:
+            temp = df_test_1
+        if round == 2:
+            temp = df_test_2
+        temp = temp[temp['results'] == type]
+        temp = temp.groupby('id')['test_date'].agg(min).reset_index() #Retrieve the minimum positive day
+        temp = temp.rename(columns = {'test_date':'first_'+ type})
+        temp['Round'] = round
+        temp_m = pd.concat([temp_m,temp],axis = 0,ignore_index=True)
+    df = df.merge(temp_m,how = 'left',on=['id','Round'])
+
+```
+
+```python
+#Create the PCR results
+df['pcr_results'] = 'Neg'
+df.loc[~df['RdRp'].isnull() | ~df['E'].isnull(),'pcr_results'] = 'Pos' 
+```
+
+```python
+#Assign the Test date
+df.loc[df['pcr_results'] == 'Pos','test_date'] = df['first_Pos']
+df.loc[df['pcr_results'] == 'Neg','test_date'] = df['first_Neg']
+
+  
+```
+
+```python
+#Retrieve the first positive date
+df_test_pos = df_test[df_test['results'] == 'Pos']
+df_test_pos = df_test_pos.groupby('id')['test_date'].agg(min).reset_index()
+df_test_pos = df_test_pos.rename(columns = {'test_date' : 'first_pos_date_overall'})
+df = df.merge(df_test_pos,on = 'id',how = 'left')
+```
+
+```python
+#Try to create the Reference date.
+df['reference_type'] = 'first_pos'
+df.loc[~df['first_symptoms_date'].isnull(),'reference_type'] = 'symptom'
+
+```
+
+```python
+#Create the numeric time
+df.loc[df['reference_type'] == 'first_pos','time'] = (df['test_date'] - df['first_pos_date_overall']).dt.days
+df.loc[df['reference_type'] == 'symptom','time'] = (df['test_date'] - df['first_symptoms_date']).dt.days
+
+```
+
+```python
+#Exclude missing time.
+df = df.loc[~df['time'].isnull(),]
+#Change the coding of negative pcr values.
+for i in ['RdRp','E']:
+    df.loc[df[i].isnull(),i] = 'negative'
+```
+
+```python
+#Enter the yaml writing stage
+participant_list = []
+for type in ['first_pos','symptom']:
+      df_output = df[df['reference_type'] == type]    
+      temp = [dict(attributes=dict(age=df_output.loc[df_output.loc[df_output["id"]==i].index[0],'age_group'].astype('object'),
```

Comment:
What are your thoughts on reporting `age_group` instead of `age` here? People might expect the age to be the exact age in years rather than the beginning of an age group. This could be particularly important for pediatric patients.

Reply:
That makes sense. Sorry, I think I told Haisu to use the average. @tillahoffmann Do we need to modify the schema for this?

Reply:
If we include the attribute as `age_group`, we won't need to modify the schema. If we want to add `0-17` to the `age` field, we'll have to make a small update. What do you think is better?

Reply:
I would recommend creating `age_group` and keeping `age` as numeric, which could be less confusing for users with less programming background.

Reply:
Age group were kept as original value, for example, 61-70.